Enhance Documentation: Clarifications and Helpers for Graphical Booting in QEMU

camilamacedo86 opened the pull request:
## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

doanac commented:
Docs for 6a396ce5f52e741dd32267be140e72bc1f735311 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2373/docs/artifacts/html/index.html

vanmaegima commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -95,7 +95,18 @@ Booting in QEMU
 -------------------
 
-In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/Weston), add the following flags to the QEMU CLI:
+In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/Weston).
+Following flags to the QEMU CLI and a general guideline:
+
+To check the available options in your system use the flag ``-display help``, e.i.:
+
+.. parsed-literal::
+   $ qemu-system-aarch64 -display help
+   Available display backend types:
+   none
+   curses
+   cocoa
+   dbus
```
I feel this sentence is lacking a verb, I would suggest something like:
```suggestion
This section shows a general guideline to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/Weston).

To check the available options in your system use the flag ``-display help``, i.e.:

.. parsed-literal::
   $ qemu-system-aarch64 -display help
   Available display backend types:
   none
   curses
   cocoa
   dbus

In general, adding the following flags to the QEMU CLI should work:
```
I'm +1 in changing this example to use ``-display default`` if we feel this works better across most architectures.

vanmaegima reviewed (commented):
I left some comments, feel free to reach out in case you want to discuss any of the topics.

camilamacedo86 commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -111,7 +122,27 @@ In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/W
+
+   Remember, specifying display options with a console-only image in QEMU doesn't convert
+   it into a graphical interface. It simply opens a window for any graphical output the
+   image might have had. Interaction remains primarily text-based.
```
I update this one let's see

doanac commented:
Docs for 4361346a68179e0514bf67a3661ddc56536b7a0a are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2376/docs/artifacts/html/index.html

camilamacedo86 commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -111,7 +122,27 @@ In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/W
+
+   Therefore, if you are following the guidance :ref:`gs-flash-device` and download the console image,
+   please revisit the available artifacts. Ensure you download an image with an interface, typically
+   prefixed with ``lmp-factory-image``.
```
It was a draft write before we figure out.

doanac commented:
Docs for c2285628cf9e5dac7d6deff63e95383b84637325 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2377/docs/artifacts/html/index.html

doanac commented:
Docs for 95d4b2779af397318610e8f34363ab80757c3beb are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2378/docs/artifacts/html/index.html

doanac commented:
Docs for 621bb31c8551cc8c5f27611c3614906518e7c53b are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2379/docs/artifacts/html/index.html

camilamacedo86 commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -111,7 +122,28 @@ In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/W
+
+   Specifying display options in QEMU doesn't make console-only images graphical. It just opens a potential graphical
+   output window. Interaction is mainly text-based. Only ``DISTRO=lmp-xwayland`` or ``DISTRO=lmp-wayland``
+   images have graphical interfaces.
```
done

doanac commented:
Docs for 3149abdd21b5fcdb7afad469c690468d56f6e6b4 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2382/docs/artifacts/html/index.html

camilamacedo86 commented on source/reference-manual/linux/linux-distro.rst:
```diff
@@ -48,11 +48,53 @@ LmP MFGTools
+Building Images With Distros
+*******************************************************************
+
+Following an example to let you know to build new targets with those images:
+
+1. Update your factory config:
+
+.. code-block:: shell
+
+    git clone https://source.foundries.io/factories/<factory>/ci-scripts.git
+    cd ci-scripts
+    vi factory-config.yml
+
+2. Add the distro under `LmP`:
+
+.. code-block:: yaml
+
+    worker_tag: "amd64-partner-aws"
+    params:
+        IMAGE: lmp-factory-image
+        SSTATE_CACHE_MIRROR: ""
+        DOCKER_COMPOSE_APP: "1"
+        DISTRO: "lmp-xwayland"
+
+3. Commit and push the changes:
+
+.. code-block:: shell
+
+    git add .
+    git commit -m "Adding lmp-xwayland as base for my factory"
+    git push
+
+4. Changes into the ``ci-script`` repository don't auto-trigger builds. Push a commit in ``lmp-manifest`` or ``meta-subscriber-overrides``:
+
+.. code-block:: shell
+
+    git clone https://source.foundries.io/factories/<factory>/lmp-manifest.git
+    cd lmp-manifest
+    git commit -m "trigger build" --allow-empty
```
This section has other tutorials as well, see:
![Screenshot 2023-10-02 at 17 04 27](https://github.com/foundriesio/docs/assets/7708031/20b67393-04c6-47f1-83fc-bfbb110586e5)

Could we merge this one here and after if we find a better place we just move it out?

angolini replied:
I believe we need a better "starting point" for customizing anything related to the factory source code. Not only changing distros, but also all the other variables. My point is, if you missed how to deal with distros, a reader who is using de default distro, but wants to change the target image or the GPLv3 packages installation default, would not know how.

whether it's called "factory customization" or "lmp customization" we can discuss.

kprosise replied:
@angolini, yes, it definitely goes into the user-guide. It is not a general use case (though it is likely common) so it may not be quiet the perfect fit in the tutorial, but wouldn't be bad either. 

 @camilamacedo86 I currently have a Jira task open to move some of the content found  under `reference-manual/linux/` to subsections/pages under `/user-guide/lmp-customization/`

So I would place it there, if @angolini agrees.

angolini replied:
I like it there @kprosise , I agree

camilamacedo86 commented:
Hi @angolini,

I've addressed most of your suggestions. Thanks for your input, especially regarding creating a new section for guidance on changing the distro.

I suggest we merge the content as it is presented in this PR. If we later identify a more suitable location for the information, we can simply move it in a subsequent update. Does that sound reasonable to you? Can we proceed with this approach?

kprosise commented on source/reference-manual/linux/linux-distro.rst:
```diff
@@ -48,11 +48,53 @@ LmP MFGTools
 
 .. _Poky: https://www.yoctoproject.org/software-overview/reference-distribution/
+
+Building Images With Distros
```
@angolini I like "Customizing the Distro" by the way. As for the glossary...Oy. It isn't really functional, and needs to be redone and really expanded on, ideally after we make a decision regarding the Static Site Generator. We _could_ add it now, which would make it easier to add later, but there is still the question of if we should add it...If it adds clarity and helps avoid confusion, go for it.

kprosise commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -95,7 +95,18 @@ Booting in QEMU
 -------------------
 
-In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/Weston), add the following flags to the QEMU CLI:
+This section shows a general guideline to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/Weston).
```
```suggestion
This section shows how to boot QEMU with an OpenGL capable virtual GPU—required for Wayland/Weston.
```

Optional suggestions,  shortening sentence.

kprosise commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -95,7 +95,18 @@ Booting in QEMU
 
-In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/Weston), add the following flags to the QEMU CLI:
+This section shows a general guideline to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/Weston).
+To check the available options in your system use the flag ``-display help``, i.e.:
```
```suggestion
To check the available options for your system, use the ``-display help`` flag:
```

kprosise commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -111,7 +122,28 @@ In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/W
 
+.. note::
+
+    You can utilize the option ``-display default``
+       It attempts to use the default display type corresponding to the operating system on which you are running the command.
```
```suggestion
    You can also utilize the option ``-display default``.
    This attempts to use the default display type corresponding to the operating system on which you are running the command.
```

kprosise commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -111,7 +122,28 @@ In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/W
 
+.. important::
+
+   Specifying display options in QEMU doesn't make console-only images graphical. It just opens a potential graphical
```
```suggestion
   Specifying display options in QEMU does not make console-only images graphical. It just opens a potential graphical
```

kprosise commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -111,7 +122,28 @@ In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/W
+   output window. Interaction is mainly text-based. The images created with ``DISTRO=lmp-xwayland`` or ``DISTRO=lmp-wayland``
+   have graphical interfaces.
+
+   If you are following :ref:`gs-flash-device` and downloading the console image, it won't boot with a
```
```suggestion
   If you are following :ref:`gs-flash-device` and downloading the console image, it will not boot with a
```

kprosise commented on source/reference-manual/qemu/qemu-instructions.template:
```diff
@@ -111,7 +122,28 @@ In order to boot QEMU with an OpenGL capable virtual GPU (required for Wayland/W
+
+   If you are following :ref:`gs-flash-device` and downloading the console image, it won't boot with a
+   graphical interface. See :ref:`LmP Wayland and LmP XWayland <ref-lmp-wayland-xwayland>` to build a new
+   target image. Also, ensure that you download the image with the interface, which usually is
```
```suggestion
   target image. Also, ensure that you download the image with the interface, which is usually
```

kprosise reviewed (commented):
Left some feedback/suggestions. I will give it another pass later. Thanks for all this work, it is really appreciated.

camilamacedo86 commented:
Hi @angolini and @kprosise 

I tried to address all suggestions.
Please, feel free to let me know if any further changes should be made.
Btw, I like the suggestion about the place to put the small guidance to help out people.

camilamacedo86 commented:
Hi @angolini,

> I just realized it could be tough to understand how to change the Distro (something that, for me, is trivial). 

I could figure it out but I really think that is hard for someone out of context to know exactly where to set up the DISTRO
that when pushing the commit for ci-script it will not build a target etc.

kprosise commented on source/user-guide/lmp-customization/lmp-customization.rst:
```diff
@@ -12,6 +12,49 @@ and other options related to the platform build.
+    cd ci-scripts
+    vi factory-config.yml
+
+2. Add the distro under `LmP`:
```
```suggestion
2. Add the distro under ``LmP``:
```

kprosise reviewed (approved):
one last little suggestion, but otherwise it looks good!

MrCry0 commented on source/reference-manual/linux/linux-distro.rst:
```diff
@@ -48,8 +51,10 @@ LmP MFGTools
+
 LmP Wayland and LmP XWayland
-****************************
+*********************************
```
This change seems unneeded. IIRC the "underline" should be the same length as the underlined text.

MrCry0 reviewed (approved):
LGTM
